File: packages/rules-store/src/rules/heating-split.ts
import type { RuleSet } from '../store';

/**
 * PLACEHOLDER SAMPLE — demonstrates the rule-set shape only.
 *
 * The real HKVO consumption/base-cost split bounds (§ 7 HeizkostenV: 50–70 %
 * consumption-based) are entered and verified at M2, together with the CO₂
 * 10-step table (CO2KostAufG). Do not treat these values as legally verified.
 */
export interface HeatingSplitBounds {
  readonly minConsumptionShare: number;
  readonly maxConsumptionShare: number;
}

export const HEATING_SPLIT_BOUNDS: RuleSet<HeatingSplitBounds> = {
  key: 'hkvo.heating-split-bounds',
  versions: [
    {
      validFrom: '1989-03-01',
      source: '§ 7 Abs. 1 HeizkostenV (PLACEHOLDER — verify at M2)',
      value: { minConsumptionShare: 0.5, maxConsumptionShare: 0.7 },
    },
  ],
};

File: packages/rules-store/src/lokara_rules_store/rules/heating_split.py
"""§ 7 Abs. 1 HeizkostenV — base/consumption split of heating cost.

At least 50 % and at most 70 % of the cost must be allocated by measured
consumption; the remainder by a fixed key (living/heated area). 70/30 is the
common default; the chosen share is per-building configuration within these
bounds.
"""

from datetime import date
from decimal import Decimal

from lokara_domain import HeatingSplitBounds

from ..store import RuleSet, RuleVersion

DEFAULT_CONSUMPTION_SHARE = Decimal("0.7")

HEATING_SPLIT_BOUNDS: RuleSet[HeatingSplitBounds] = RuleSet(
    key="hkvo.heating-split-bounds",
    versions=(
        RuleVersion(
            valid_from=date(1989, 3, 1),
            source="§ 7 Abs. 1 HeizkostenV",
            value=HeatingSplitBounds(
                min_consumption_share=Decimal("0.5"),
                max_consumption_share=Decimal("0.7"),
            ),
        ),
    ),
)
